elf/pinning.go

```diff
@@ -67,3 +76,24 @@ func pinObject(fd int, namespace, object, name string) error {
 func PinObjectGlobal(fd int, namespace, name string) error {
 	return pinObject(fd, namespace, BPFDirGlobals, name)
 }
+
+func PinObjectCustom(fd int, customPath, name string) error {
+	mounted, err := bpffs.IsMounted()
+	if err != nil {
+		return fmt.Errorf("error checking if %q is mounted: %v", BPFFSPath, err)
+	}
+	if !mounted {
+		return fmt.Errorf("bpf fs not mounted at %q", BPFFSPath)
+	}
+
+	if customPath == "" {
+		return fmt.Errorf("no custom path is given")
+	}
+	customHash := hex.EncodeToString(util.MakeSha256([]byte(customPath)))
```

Comment:
I think it should not be up to gobpf to decide whether the path should be hashed or not. The caller of the gobpf library should decide that. Some users of gobpf might use cgroup path, but other users (such as iproute2's tc) might not.